src/reducers/polling.js

```diff
@@ -299,7 +305,11 @@ export const pollsInit = () => async dispatch => {
           dispatch(addPoll(pollData));
           dispatch(pollDataInit(pollData));
         })
-        .catch(e => console.error(e));
+        .catch(e => console.error(e))
+        .finally(_ => {
+          pollsRemaining--;
+          if (pollsRemaining === 0) dispatch(pollsSuccess());
```

Comment:
@jparklev seems a little bit weird to do this with a counter, but otherwise we don't know when exactly polls are finished fetching. If we only check the for the `poll` to exist before showing `NotFound`, the component appears for about a second before `poll` populates and the page loads as normal.

Reply:
Right, maybe one of the drawbacks of firing off promises like this. I've seen some version of this before and it seems like a fair thing to do in this situation.

I think we might be able to create scheme where we cache the polls from `getAllWhiteListedPolls` immediately giving each poll a `fetching` flag that's flipped when the promise associated w/ that poll resolves. Then, the component can check something like `poll list has been populated` & `multihash from url is in the list` & `poll successfully fetched`. But yea, ¯\_(ツ)_/¯ what you have here is cool w/ me